lib/src/Uncertainty/Algorithm/OrthogonalBasis/openturns/OrthogonalProductFunctionFactory.hxx

```diff
@@ -62,12 +62,18 @@ public:
   /** Build the Function of the given index */
   Function build(const UnsignedInteger index) const override;
 
+  /** Build the Function of the given multi-indices */
+  Function build(const Indices & indices) const override;
+
   /** Return the enumerate function that translate unidimensional indices nto multidimensional indices */
   EnumerateFunction getEnumerateFunction() const override;
 
   /** Return the collection of univariate orthogonal polynomial families */
   FunctionFamilyCollection getFunctionFamilyCollection() const;
 
+  /** Get marginal functions */
+  TensorizedUniVariateFunctionFactory::FunctionFamilyCollection getMarginal(const Indices & indices) const;
```

Comment:
its not obvious that this one is needed (and the one for the polynoms)
I dont see where its actually used, and also I think it should be part of the generic api in the parent class
but for that build(Indices) can play the same role, so I think this one should be removed (and the one for the polynoms)

Reply:
We need this for:
https://github.com/openturns/openturns/pull/2372/files#diff-c8a311f20d667afa10f5da5737cbc0bda64e76990b117082d9cbd21245d3f58e

In this code, we need OrthogonalProductPolynomialFactory.getMarginal() in the case where the factories are polynomials and OrthogonalProductFunctionFactory.getMarginal() in the general, function, case.

Reply:
Please declare `getMarginal()` to the `OrthogonalFunctionFactory`. Then override these methods at the lower levels. For `OrthogonalFunctionFactory`, throw a "not yet implemented".

Reply:
That is not possible. OrthogonalFunctionFactory is based on a single function, while OrthogonalProductPolynomialFactory and OrthogonalProductFunctionFactory are based on a collection of functions. Hence, the type returned by the hypothetical OrthogonalFunctionFactory.getMarginal() must be a single factory, which does not fit the type required by the two children classes, which is a collection of factories.